JT/12_data_structures/linear_types_exercises/linear_types.adoc

```diff
@@ -0,0 +1,238 @@
+:icons: font
+== Array
+
+image::../resources/array.png[align="center",width=700]
+
+An Array is a data structure used to collect multiple data elements of the same data type into one variable. +
+
+*Arrays can be classified into different types:*
+
+* *One-Dimensional Array:* An Array with only one row of data elements is known as a One-Dimensional Array.
+It is stored in ascending storage location.
+* *Two-Dimensional Array:* An Array consisting of multiple rows and columns of data elements is called a Two-Dimensional Array.
+It is also known as a Matrix.
+* *Multidimensional Array:* We can define Multidimensional Array as an Array of Arrays.
+Multidimensional Arrays are not bounded to two indices or two dimensions as they can include as many indices are per the need. +
+
+.*How to Create the Array?*
+----
+dataType[] arrayName;
+double[] data;
+----
+
+.*How many elements can array this hold?*
+To define the number of elements that an array can hold, we have to allocate memory for the array in Java.
+In Java, we can declare and allocate the memory of an array in one single statement.
+For example,
+
+----
+double[] data = new double[10];
+----
+
+.*How to Initialize Array?*
+Here, we have created an array named age and initialized it with the values inside the curly brackets.
+
+Note that we have not provided the size of the array.
+In this case, the Java compiler automatically specifies the size by counting the number of elements in the array.
+
+----
+//declare and initialize and array
+int[] age = {12, 4, 5, 2, 5};
+----
+
+[NOTE]
+====
+* Array indices always start from 0. That is, the first element of an array is at index 0.
+* If the size of an array is n, then the last element of the array will be at index n-1.
+====
+
+.*How to Access Elements of an Array?*
+We can access the element of an array using the index number.
+
+----
+// access array elements
+array[index]
+----
+
+Multidimensional Arrays +
+A multidimensional array is an *array of arrays*.
+That is, each element of a multidimensional array is an array itself.
+For example,
+
+----
+double[][] matrix = {{1.2, 4.3, 4.0},
```

Comment:
Please fix arrangement of the arrays. It could have better view like:
```
double[][] matrix = {
        {1.2, 4.3, 4.0},
        {4.1, -1.1}
};
```

Reply:
Done